### PR TITLE
Allow to protect shares by link with a password

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: web/apps/serve.go [hunk 1]
```diff
@@ -248,17 +248,17 @@ func ServeAppFile(c echo.Context, i *instance.Instance, fs appfs.FileServer, web
 				return renderMovedLink(c, i, to, subdomainType)
 			}
 		}
+	}
```
Comment (taratatach) [inline web/apps/serve.go:251]: Shouldn't logged-in user have direct access to their password-protected sharings instead?

File: assets/scripts/share-by-link.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const submit = d.getElementById('password-submit')
+  const input = d.getElementById('password')
+  const permID = d.getElementById('perm-id')
+
+  const onSubmit = function (event) {
+    event.preventDefault()
+    input.setAttribute('disabled', true)
+    submit.setAttribute('disabled', true)
+
+    const data = new URLSearchParams()
```
[the remark this answers is not in the document]
Comment (nono) [reply assets/scripts/share-by-link.js:15]: My internet was down, and I have copied that code from another JS script

File: web/apps/serve.go (hunk 1 above)
Comment (nono) [reply web/apps/serve.go:251]: No, the user may want to check that the link they will give is really protected by a password. And they can still access their data via the normal way.

File: assets/scripts/share-by-link.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nono) [reply assets/scripts/share-by-link.js:15]: https://developer.mozilla.org/en-US/docs/Web/API/FormData says:

> It uses the same format a form would use if the encoding type were set to "multipart/form-data"

We don't want `multidata/form-data` but `application/ x-www-form-urlencoded`, and it looks `URLSearchParams` is the good way to do that in JS.

**Review by taratatach — APPROVED** [review]

**Pull request merged**